src/Interpreters/InterpreterSelectWithUnionQuery.cpp

```diff
@@ -231,6 +231,17 @@ Block InterpreterSelectWithUnionQuery::getSampleBlock(const ASTPtr & query_ptr_,
         return cache[key] = InterpreterSelectWithUnionQuery(query_ptr_, context_, SelectQueryOptions().analyze()).getSampleBlock();
 }
 
+/// Daisy : starts
+bool InterpreterSelectWithUnionQuery::hasAggregation() const
+{
+    for (auto & interpreter : nested_interpreters)
+    {
+        if (interpreter->hasAggregation())
+            return true;
+    }
+    return false;
+}
+/// Daisy : ends
```

Comment:
`SelectQuery::hasAggregation` used on src/Storages/ProjectionsDescription.cpp:145,
but where is `InterpreterSelectWithUnionQuery::hasAggregation` used and base class `IInterpreterUnionOrSelectQuery::hasAggregation` used ?

Reply:
"InterpreterSelectWithUnionQuery::hasAggregation used and base class IInterpreterUnionOrSelectQuery::hasAggregation used", hasAggregation method is new to these two classes, and this method is only used in our code in "SQLAnalyzerRestRouterHandler.cpp"